File: AIEP-Plugin-FIR/src/com/njucs/aiep/plugin/fir/ai/FIR_AI.java
package com.njucs.aiep.plugin.fir.ai;

import com.njucs.aiep.game.GameTree;
import com.njucs.aiep.game.Status;
import com.njucs.aiep.net.DataTransmit.StudentInfo;
import com.njucs.aiep.plugin.AI;
import com.njucs.aiep.plugin.fir.frame.Step;


/**
 * 
 * @author ygsx
 * 
 * @time 2013Äê5ÔÂ5ÈÕ23:59:26
 * */
public interface FIR_AI extends AI {

	public Step itsmyturn( Step opponentStep );
	public Step itsmyturn2( Step opponentStep );
	
	public void setInningInfo( Status myStatus, int limitedTime, 
			StudentInfo opponentInfo, final Status[] piecesArray);
	
	public GameTree getLastGameTree();
	
}

File: AIEP-C-Set-Client/AIEP-CSharp-Client/FIR_AICS/FIR_AI.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using System.Runtime.InteropServices;

namespace AICS
{
    /// <summary>
    /// FIR(Fice In Row) AI interface
    /// </summary>
    public interface FIR_AI : AI
    {
        /// <summary>
        /// Obtain the  step of the current AI located.<br />
        /// The first heuristic evaluation function.  <br />This method will be called when 
        ///  the server chooses the "Heriustic1" Item.
        /// </summary>
        /// <param name="opponentStep">the last step of the opponent located</param>
        /// <returns>the new step</returns>
        Step itsmyturn(Step opponentStep);

        /// <summary>
        /// Obtain the  step of the current AI located.<br />
        ///  The second heuristic evaluation function.<br /> This method will be called when 
        ///  the Judge chooses the "Heriustic2" Item.
        /// </summary>
        /// <param name="opponentStep">the last step of the opponent located</param>
        /// <returns>the new step</returns>
        Step itsmyturn2(Step opponentStep);

        /// <summary>
        /// Set the inning info. <br />When the data transmitter receives an info message, it will be called!
        /// </summary>
        /// <param name="myStatus">the current status</param>
        /// <param name="limitedTime">the limited time</param>
        /// <param name="opponentInfo">the opponent info</param>
        /// <param name="piecesArray">the initial inning array, length = 15*15 </param>
        void setInningInfo(Status myStatus, int limitedTime, StudentInfo opponentInfo,
            Status[] piecesArray);


        /// <summary>
        /// Get the game tree generated at the last step
        /// </summary>
        /// <returns>the game tree root generated at the last step</returns>
        GameTree getLastGameTree();
    }





}
